### PR TITLE
ESLint: custom directories from next.config.js

### LINKED ISSUE #26393: ESLint does not use custom directories from next.config.js
### What version of Next.js are you using?

11.0.0

### What version of Node.js are you using?

16.1.0

### What browser are you using?

not important

### What operating system are you using?

macOS

### How are you deploying your application?

not important

### Describe the Bug

[Custom directories for linting](https://nextjs.org/docs/basic-features/eslint#linting-custom-directories) are not used when running `next lint`, but only when running `next build`

We are missing eslint.dirs somewhere:
https://github.com/vercel/next.js/blob/0ca24563be9658732eac158c31046ff58bfe78ca/packages/next/cli/next-lint.ts#L133-L142

as we have it here:
https://github.com/vercel/next.js/blob/0ca24563be9658732eac158c31046ff58bfe78ca/packages/next/build/index.ts#L217

Related to: https://github.com/vercel/next.js/pull/25895

### Expected Behavior

Custom directories for linting are used also on `next lint` run.

### To Reproduce

Create project and configure any custom directory in `next.config.js`, put there any file with some obvious issues and try to run linter.

It's not going to fail with `next lint`, but only with `next build`.

### PR DESCRIPTION
**Pull request by michalbundyra**
Attempt to fix #26393

~~Sorry, I wasn't able to run tests on local, but I hope at least test should be valid.~~
Edit: Tests are working correctly on my local :) 

fixes #26393

## Bug

- [x] Related issues linked using `fixes #26393`
- [x] Integration tests added

### CODE REVIEW

File: packages/next/cli/next-lint.ts [hunk 1]
```diff
@@ -131,7 +131,9 @@ const nextLint: cliCommand = (argv) => {
     printAndExit(`> No such directory exists as the project root: ${baseDir}`)
   }
 
-  const dirs: string[] = args['--dir']
+  const conf = await loadConfig(PHASE_PRODUCTION_BUILD, baseDir)
+
+  const dirs: string[] = args['--dir'] ?? conf.eslint?.dirs
```
Comment (housseindjirdeh) [inline packages/next/cli/next-lint.ts:136]: Could you also update the [ESLint docs](https://github.com/vercel/next.js/blob/2a5db0b58c9416381568b9716861b598b17a7015/docs/basic-features/eslint.md#linting-custom-directories) to remove the mention that this only works for production builds? It's probably still worth keeping the `--dir` flag, so could we also include a line on precedence? (i.e. Using `--dir` will take precedence over `conf.eslint?.dirs`)

**Review by housseindjirdeh — CHANGES_REQUESTED** [review]
Thanks for this @michalbundyra. The workaround for `next lint` was to use the `--dir` flag instead, but I agree that it's probably more intuitive that `config.eslint.dirs` also works the same for both `next lint` and `next build` :)

One comment around docs, otherwise looks good!

One tiny

**Comment by michalbundyra** [comment]
@housseindjirdeh thanks for the review. Documentation is now updated and branch rebased. Thanks

File: docs/basic-features/eslint.md [hunk 2]
```diff
@@ -54,22 +54,24 @@ If you do not want ESLint to run as a build step, refer to the documentation for
...
 ```
 
-Similarly, the `--dir` flag can be used for `next lint`:
+Also, the `--dir` flag can be used for `next lint`:
 
 ```bash
 yarn lint --dir pages --dir utils
 ```
 
+and it will take precedence over the configuration from `next.config.js`.
```
Comment (housseindjirdeh) [inline docs/basic-features/eslint.md:73]: Tiny nit :)

```suggestion
This will take precedence over the configuration from `next.config.js`.
```

**Review by housseindjirdeh — DISMISSED** [review]
Thanks, LGTM!

**Review by leerob — DISMISSED** [review]
Docs look okay!

**Comment by bnjmnt4n** [comment]
Hi, just wanted to check is there anything blocking this? The current behaviour was a little confusing, and I'd look forward to seeing this merged.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| buildDuration | 18s | 17.9s | -107ms |
| buildDurationCached | 4s | 3.9s | -136ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +301 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.076 | 3.103 | ⚠️ +0.03 |
| / avg req/sec | 812.68 | 805.73 | ⚠️ -6.95 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.915 | 2.018 | ⚠️ +0.1 |
| /error-in-render avg req/sec | 1305.44 | 1238.97 | ⚠️ -66.47 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| 359.HASH.js gzip | 2.96 kB | 2.96 kB | ✓ |
| 745.HASH.js gzip | 180 B | 180 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 21 kB | 21 kB | ✓ |
| webpack-HASH.js gzip | 1.53 kB | 1.53 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| _app-HASH.js gzip | 803 B | 803 B | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 554 B | 554 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.52 kB | 2.52 kB | ✓ |
| head-HASH.js gzip | 2.28 kB | 2.28 kB | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| image-HASH.js gzip | 5.63 kB | 5.63 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 19.1 kB | 19.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| _buildManifest.js gzip | 489 B | 489 B | ✓ |
| Overall change | 489 B | 489 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 542 B | 542 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| buildDuration | 13.9s | 13.9s | ⚠️ +2ms |
| buildDurationCached | 5.5s | 5.4s | -74ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +301 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.102 | 2.993 | -0.11 |
| / avg req/sec | 805.8 | 835.32 | +29.52 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.916 | 1.892 | -0.02 |
| /error-in-render avg req/sec | 1304.57 | 1321.33 | +16.76 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| 17.HASH.js gzip | 2.98 kB | 2.98 kB | ✓ |
| 18.HASH.js gzip | 185 B | 185 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 8.4 kB | 8.4 kB | ✓ |
| webpack-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| Overall change | 68.5 kB | 68.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| _app-HASH.js gzip | 791 B | 791 B | ✓ |
| _error-HASH.js gzip | 3.76 kB | 3.76 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.02 kB | 3.02 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.6 kB | 17.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | michalbundyra/next.js hotfix/eslint-custom-directories | Change |
| - | - | - | - |
| index.html gzip | 576 B | 576 B | ✓ |
| link.html gzip | 588 B | 588 B | ✓ |
| withRouter.html gzip | 569 B | 569 B | ✓ |
| Overall change | 1.73 kB | 1.73 kB | ✓ |

</details>

</details>
Commit: 831367686360ea679ec93303d94241ebe99cc2f2

**Review by ijjk — APPROVED** [review]
Thanks for the PR!

**Comment by ijjk** [comment]
Hi, this is now available in `v11.0.2-canary.22` of Next.js, please update and give it a try!

**Pull request merged**